client/components/Modals/AmenitiesModal.jsx

```diff
@@ -18,48 +65,44 @@ const AmenitiesModal = ({ toggleModal, listItems }) => {
     document.addEventListener('mousedown', handleClickOutside);
     return () => document.removeEventListener('mousedown', handleClickOutside);
   });
+
+  // Creating a map of subheaders
+  let mapOfHeaders = {};
+  headers.forEach((header) => {
+    // Initializing each value to an array
+    mapOfHeaders[header] = [];
+  });
+  // Pushing each amenity into its respective subheader
+  listItems.map((item) => {
+    mapOfHeaders[item.subheader].push(item);
+  });
+
+  let organizedHeaders = [];
   if (listItems.length > 0) {
-    itemsToList = listItems.map((item) => {
-      return (
-        <li>
-          {item.title}
-          {item.amenityDescription}
-          {item.subHeader}
+    // Looping thru the map with each subHeader
+    for (let item in mapOfHeaders) {
+      // Pushing the respective subHeader into the organizedHeaders array
+      organizedHeaders.push(
+        <li style={ListItemStyle}>
+          <b> {item}</b>
         </li>
       );
-    });
+      // Pushing the children of THAT subheader into the organizedHeaders array
```

Comment:
love that you're using high order functions!

Reply:
Yessss. I'm trying to move away from for loops whenever possible!!